graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/TopSecondDegreeByCount.java

```diff
@@ -61,96 +43,109 @@
    *
    * @param leftIndexedBipartiteGraph is the
    *                                  {@link NodeMetadataLeftIndexedMultiSegmentBipartiteGraph}
-   *                                  to run TopSecondDegreeByCount on
+   *                                  to run TopSecondDegreeByCountForTweet on
    * @param expectedNodesToHit        is an estimate of how many nodes can be hit in
-   *                                  TopSecondDegreeByCount. This is purely for allocating needed
+   *                                  TopSecondDegreeByCountForTweet. This is purely for allocating needed
    *                                  memory right up front to make requests fast.
    * @param statsReceiver             tracks the internal stats
    */
   public TopSecondDegreeByCount(
-    NodeMetadataLeftIndexedMultiSegmentBipartiteGraph leftIndexedBipartiteGraph,
-    int expectedNodesToHit,
-    StatsReceiver statsReceiver
-  ) {
+      NodeMetadataLeftIndexedMultiSegmentBipartiteGraph leftIndexedBipartiteGraph,
+      int expectedNodesToHit,
+      StatsReceiver statsReceiver) {
     this.leftIndexedBipartiteGraph = leftIndexedBipartiteGraph;
-    this.visitedRightNodes = new Long2ObjectOpenHashMap<NodeInfo>(expectedNodesToHit);
-    this.nodeInfosAfterFiltering = new ArrayList<NodeInfo>();
+    this.visitedRightNodes = new Long2ObjectOpenHashMap<>(expectedNodesToHit);
+    this.nodeInfosAfterFiltering = new ArrayList<>();
     this.seenEdgesPerNode = new Long2ByteArrayMap();
     this.topSecondDegreeByCountStats = new RecommendationStats();
     this.statsReceiver = statsReceiver.scope("TopSecondDegreeByCount");
     this.numRequestsCounter = this.statsReceiver.counter("numRequests");
   }
 
-  private void collectRecommendations(TopSecondDegreeByCountRequest request) {
+  /**
+   * Interface method for updating information gathered about each RHS node.
+   * @param leftNode                is the LHS node from which traversal initialized
+   * @param rightNode               is the RHS node at which traversal arrived
+   * @param edgeType                is the edge from which LHS and RHS nodes are connected
+   * @param weight                  is the weight contributed to a RHS node in this traversal
+   * @param edgeIterator            is the iterator for traversing edges from LHS node
+   * @param maxSocialProofTypeSize  is the maximum social proof types to keep
+   * @param collectedRightNodeInfo  is a map of visited nodes, containing their info such as weights
+   */
+  protected abstract void updateNodeInfo(
+      long leftNode, long rightNode, byte edgeType, double weight,
```

Comment:
1) One variable per line.

2) I know there is some conflict between style guide and GraphJet code base, but I would keep the indentation to 2 spaces everywhere.